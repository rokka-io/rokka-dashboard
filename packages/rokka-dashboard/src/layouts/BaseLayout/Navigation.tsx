import React, { FunctionComponent } from 'react';
import { LiPosRelative } from 'rokka-dashboard-ui-base';
import { navigation } from '../../pages/routes';

export const Navigation: FunctionComponent = () => (
  <>
    {navigation.map(({ name, relative, navigationComponent }) =>
      relative ? <LiPosRelative key={name}>{navigationComponent}</LiPosRelative> : <li key={name}>{navigationComponent}</li>
    )}
  </>
);
