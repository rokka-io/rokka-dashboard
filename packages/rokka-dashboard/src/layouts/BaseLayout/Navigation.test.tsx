import { shallow } from 'enzyme';
import React from 'react';
import { Navigation } from './Navigation';

describe('Compositions', () => {
  describe('Navigation', () => {
    it('renders the Navigation correctly.', () => {
      const component = shallow(<Navigation />);

      expect(component).toMatchSnapshot();
    });
  });
});
