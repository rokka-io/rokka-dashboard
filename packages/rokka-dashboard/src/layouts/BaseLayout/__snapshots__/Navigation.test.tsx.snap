// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compositions Navigation renders the Navigation correctly. 1`] = `
<Fragment>
  <li
    key="Dashboard"
  >
    <Component
      exact={true}
      name="Dashboard"
      path="/"
    />
  </li>
  <li
    key="Images"
  >
    <Component
      name="Images"
      path="/images"
    />
  </li>
  <styled.li
    key="Stacks"
  >
    <Component
      newStackRoute={
        Object {
          "name": "New Stack",
          "path": "/new-stack",
        }
      }
      stacksRoute={
        Object {
          "name": "Stacks",
          "path": "/stacks",
        }
      }
    />
  </styled.li>
</Fragment>
`;
